filament/backend/include/backend/platforms/VulkanPlatform.h

```diff
@@ -21,24 +21,120 @@
 
 #include <bluevk/BlueVK.h>
 #include <utils/FixedCapacityVector.h>
+#include <utils/PrivateImplementation.h>
 
+#include <tuple>
 #include <unordered_set>
 
 namespace filament::backend {
 
+using SwapChain = Platform::SwapChain;
+
 /**
- * A Platform interface that creates a Vulkan backend.
+ * Private implementation details for the provided vulkan platform.
  */
+struct VulkanPlatformPrivate;
 
-class VulkanPlatform : public Platform {
+/**
+ * A Platform interface that creates a Vulkan backend.
+ */
+class VulkanPlatform : public Platform, public utils::PrivateImplementation<VulkanPlatformPrivate> {
 public:
+    /**
+     *
+     * Shorthand for the pointer to the Platform SwapChain struct, we use it also as a handle (i.e.
+     * identifier for the swapchain). (We didn't name this SwapChainHandle to avoid naming conflict
+     * with Handle.h).
+     */
+    using SwChainHandle = Platform::SwapChain*;
```

Comment:
should be `SwapChainHandle` (`sw` could mean `software` :))
In fact maybe it should be `SwapChaintPtr` because `SwapChainHandle` could be confused with `Handle<SwapChain>`

Reply:
renamed.